docs/src/getting_started/fit_simulation.md

```diff
@@ -167,101 +215,184 @@ u0 = [1.0, 1.0]
+The `Optimization.solve` function can accept an optional callback function
+to monitor the optimization process using extra arguments returned from `loss`.
 
-This step will look very similar to [the first optimization tutorial](@ref first_opt), except now we have a new
-cost function. Here we'll also define a callback to monitor the solution process, more details about callbacks in Optimization.jl can be found [here](https://docs.sciml.ai/Optimization/stable/API/solve/). 
-However, this time, our function returns two things. The callback syntax is always `(value being optimized, arguments of loss return)`
-and thus this time the callback is given `(p, l, sol)`. See, returning the solution along with the loss as part of the
-loss function is useful because we have access to it in the callback to do things like plot the current solution
-against the data! Let's do that in the following way:
+The callback syntax is always:
+
+```
+callback(
+    optimization function parameters,
+    value being optimized,
+    other arguments returned from the loss function, ...
+)
+```
+
+In this case, we will provide the callback the arguments `(_, l, sol)`,
+since there are no additional optimization function parameters.
+The return value of the callback function should default to `false`.
+`Optimization.solve` will halt if/when the callback function returns `true` instead.
+Typically the `return` statement would monitor the loss value
+and stop once some criteria is reached, e.g. `return loss < 0.0001`,
+but we will stop after a set number of iterations instead.
+More details about callbacks in Optimization.jl can be found
+[here](https://docs.sciml.ai/Optimization/stable/API/solve/).
 
 ```@example odefit
-callback = function (p, l, sol)
+function callback(_, l, sol)
     display(l)
-    plt = plot(sol, ylim = (0, 6), label = "Current Prediction")
-    scatter!(plt, datasol, label = "Data")
+    plt = plot(sol, ylim = (0, 6), label = ["Current x Prediction" "Current y Prediction"])
+    scatter!(plt, t_data, xy_data', label = ["x Data" "y Data"])
     display(plt)
-    # Tell Optimization.solve to not halt the optimization. If return true, then
-    # optimization stops.
     return false
 end
 ```
 
-Thus every step of the optimization will show us the loss and a plot of how the solution
-looks vs the data at our current parameters.
+With this callback function, every step of the optimization will display both the loss value and a plot of how the solution compares to the training data.
+
+Now, just like [the first optimization tutorial](@ref first_opt),
+we set up our `OptimizationFunction` and `OptimizationProblem`,
+and then `solve` the `OptimizationProblem`.
+We will initialize the `OptimizationProblem` with the same `pguess` we used
+when setting up the ODE Model in Step 3.
+Observe how `Optimization.solve` brings the model closer to the experimental data as it iterates towards better ODE parameter values!
 
```

Comment:
```suggestion

Note that we are using the `PolyOpt()` solver choice here which is discussed https://docs.sciml.ai/Optimization/dev/optimization_packages/polyopt/ since parameter estimation of non-linear differential equations is generally a non-convex problem so we want to run a stochastic algorithm (Adam) to get close to the minimum and then finish off with a quasi-newton method (L-BFGS) to find the optima.
```